### ISSUE TITLE
Replace WordPress events in dashboard with feature votes

### ISSUE THREAD

**Issue by nylen** [ClassicPress/ClassicPress#40]
Let's decide what to do about the WordPress events feed that comes through in the admin dashboard.

I think we should remove it from ClassicPress, because these are officially sponsored WordPress events, and it doesn't seem fair to encourage people to "crash the party" so to speak.

This will also eliminate some calls to the WordPress.org APIs.

**Comment by senlin** [comment]
I'm also for removing the lot.

**Comment by KTS915** [comment]
That makes sense to me too.

**Comment by scottybo** [comment]
I vote for removing - ClassicPress branding will be everywhere apart from in the events, and I think this will lead to confusion.

RE: The events themselves, I've been using WordPress for 10+ years and have **never** found this area on the dashboard useful.

**Comment by nylen** [comment]
It seems like we have a consensus here.  Let's use an API endpoint under https://api.classicpress.net for this, so that we can modify the implementation later if needed.  See:  https://github.com/ClassicPress/ClassicPress-APIs/issues/3

It would also be nice to have users search for their feature request before they submit it - we are already seeing some duplicate requests coming up on https://vote.classicpress.net.

**Comment by dsnid3r** [comment]
Forgot to comment here but I support the decision of replacing WordPress events tab in the dashboard but on a multisite install it should only show up in the Network Admin panel. The way it currently works if I recall properly is it shows up on the Network panel and the primary site, I move to strike it from the primary sites dashboard and go exclusively to the Network panel.

**Comment by dsnid3r** [comment]
Once we get lock down the confirmation of how we're moving forward with voting and our own ClassicPress website I can handle this. (already have the code ready from one of my multisites)

**Comment by nylen** [comment]
Hi @dsnid3r, let's go ahead with the implementation (removing all code related to WP events, and replacing it with ClassicPress feature requests).  Here is the first step in implementing this:

> Let's use an API endpoint under https://api.classicpress.net for this, so that we can modify the implementation later if needed. See: ClassicPress/ClassicPress-APIs#3

**Comment by dsnid3r** [comment]
> Hi @dsnid3r, let's go ahead with the implementation (removing all code related to WP events, and replacing it with ClassicPress feature requests). Here is the first step in implementing this:
> 
> > Let's use an API endpoint under https://api.classicpress.net for this, so that we can modify the implementation later if needed. See: [ClassicPress/ClassicPress-APIs#3](https://github.com/ClassicPress/ClassicPress-APIs/issues/3)

You got it.

**Comment by dsnid3r** [comment]
I will get back to this on Monday.

**Comment by dsnid3r** [comment]
Dropped the ball on this, if anyone wants to take this over that would be great.

**Comment by nylen** [comment]
Milestoning for our final release for now, as this is likely to be a significant amount of work.  Hopefully we can get it done in a beta release though!

**Comment by brettvan** [comment]
I am taking a look at this issue.   Will be at WC Philly this weekend though.

**Comment by mikeschinkel** [comment]
Somewhat related _(since they are about revamping the admin):_ 

- [Create an Admin Settings page w/icons grouped by category](https://petitions.classicpress.net/posts/101/create-an-admin-settings-page-w-icons-grouped-by-category)
- [Implement Clean URLs in the Admin](https://petitions.classicpress.net/posts/102/implement-clean-urls-in-the-admin)

**Comment by dikiyforester** [comment]
Can I help with this ticket?

**Comment by nylen** [comment]
Hi @dikiyforester, yes, definitely!

You can use this API endpoint to pull the needed data from within ClassicPress: https://api-v1.classicpress.net/features/1.0/
More info here: https://github.com/ClassicPress/ClassicPress-APIs/issues/3

**Comment by dikiyforester** [comment]
The widget "Events and News" actually includes two separated parts: News and Events.
See here https://github.com/ClassicPress/ClassicPress/blob/57705dc2ff1c1da4b83bc67b2c096bb4ec7718f6/src/wp-admin/includes/dashboard.php#L1080-L1086

Should we leave the News part or remove it along with Events? We actually can connect it to ClassicPress blog endpoint and feed.

**Comment by nylen** [comment]
> We actually can connect it to ClassicPress blog endpoint and to the RSS feed.

This sounds good to me.

**Comment by bahiirwa** [comment]
I have done a prototype plugin on https://github.com/bahiirwa/ClassicPressPetitionsDashboard Kindly check and advise if this is a step forward or backwards

**Comment by mikeschinkel** [comment]
@bahiirwa Can we get you to add some screenshots to your repo?  :-)

**Comment by nylen** [comment]
@bahiirwa thanks for doing this. I will take a look at the plugin today.

**Comment by bahiirwa** [comment]
@mikeschinkel Done with request.

**Comment by mikeschinkel** [comment]
@bahiirwa +1

**Comment by bahiirwa** [comment]
> I have done a prototype plugin on https://github.com/bahiirwa/ClassicPressPetitionsDashboard Kindly check and advise if this is a step forward or backwards

With the help of @mikeschinkel @striebwj I think, the plugin is now good for a proper review.

![screenshot-1](https://user-images.githubusercontent.com/7713923/49385555-c6c29f80-f72e-11e8-9dd4-d4e947b7404e.png)

**Comment by bahiirwa** [comment]
PR done.

**Comment by dikiyforester** [comment]
@nylen Could you please reopen since it still needs to remove Events code.
I was planning to do that in the next PR (after #268).

**Comment by bahiirwa** [comment]
Do you really want to do that, there was a request in slack to have this remain as some folks want to attend these events for learning sake. Will it be deprecated into a plugin at least?

**Comment by dikiyforester** [comment]
> Do you really want to do that, there was a request in slack to have this remain as some folks want to attend these events for learning sake. Will it be deprecated into a plugin at least?

Ohhh, I wasn't aware of this. Just following the original ticket task.
@nylen could you clarify this?

**Comment by nylen** [comment]
Reopening this issue.  I meant to reopen after I merged that PR but forgot.

I still think we should remove the rest of this feature, because WP events are not our events, and we don't want to encourage people to "crash the party" so to speak.  Instead, we want to focus on our petitions list and the news feed from our site (getting people involved in our project online).

> Will it be deprecated into a plugin at least?

I'd love to be able to do this :) However, we don't really have a good place to put this plugin right now.

If someone wants to write a plugin for this and submit it to the WP plugin directory, that would work.  In that case, probably best to write it as a plugin that can disable the events feed too, so that it has value for WP users.

**Comment by ruiguerreiro79** [comment]
what need's to be done on this one?

**Comment by nylen** [comment]
@ruiguerreiro79 the WP events feed needs to be removed from the ClassicPress dashboard until we have something similar that can replace it.

We already have blog posts from www.classicpress.net/blog coming through, as well as feature votes from the petitions site.  That is good enough for v1.

**Comment by ruiguerreiro79** [comment]
@nylen So is it ok to remove the red areas of the following image? The News at the bottom point to the classicpress.net/blog
![image](https://user-images.githubusercontent.com/29898220/52603498-027c9280-2e5f-11e9-9da8-215928738fbf.png)

**Comment by nylen** [comment]
@ruiguerreiro79 yes, remove the red boxes in your screenshot, and also change the title to just "ClassicPress News".

**Comment by nylen** [comment]
PR for this issue: #367
